ma/api/custom/maintenance.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import requests
+
+import ma.utility
+import ma.exceptions
+import ma.api.base_class
+import ma.config.magento
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MaintenanceApi(ma.api.base_class.Api):
+    def __init__(self):
+        self.__uh = ma.config.magento.UTILITY_HASH
+        self.__mmu = ma.config.magento.MAGENTO_MAINTENANCE_URL
+
+        self.__mct =  ma.config.magento.MAGENTO_CACHE_TYPES
+        self.__mit = ma.config.magento.MAGENTO_INDEX_TYPES
+
+    def __get_matching_entries(self, d, l):
+        # Returns all the dictionary entries as a tuple
+        # based on the matching keys from the passed list
+
+        # defaults to all entries when nothing to match.
+        # Used as flag to process the entire set.
+        if l is None:
+            return d.items()
+
+        return  ma.utility.match_dict_keys_from_list(d, l)
```

Comment:
I don't think one line of code merits two functions that requires six-lines and some comments, but that's my observation. It's up to you.